src/snowflake/cli/api/entities/application_package_entity.py

```diff
@@ -1,10 +1,35 @@
+from pathlib import Path
+
+from snowflake.cli._plugins.nativeapp.artifacts import build_bundle
+from snowflake.cli._plugins.nativeapp.bundle_context import BundleContext
+from snowflake.cli._plugins.nativeapp.codegen.compiler import NativeAppCompiler
+from snowflake.cli._plugins.workspace.action_context import ActionContext
 from snowflake.cli.api.entities.common import EntityBase
+from snowflake.cli.api.project.schemas.entities.application_package_entity_model import (
+    ApplicationPackageEntityModel,
+)
 
 
 class ApplicationPackageEntity(EntityBase):
     """
     A Native App application package.
     """
 
-    def bundle(self):
-        raise NotImplementedError("Application package bundle")
+    _entity_model: ApplicationPackageEntityModel
+
+    def bundle(self, ctx: ActionContext):
+        model = self._entity_model
+        bundle_map = build_bundle(
+            ctx.project_root, Path(model.deploy_root), model.artifacts
```

Comment:
can/should we make `model.deploy_root` a `Path` object? let's avoid keeping paths as raw strings as early as possible

(also for the other path-like strings in that class)

Reply:
We started with `Path`, and after a previous debate we decided to keep it as `str` on the model class for now. We can revisit this decision in the future.